Marlin/Marlin_main.cpp

```diff
@@ -4295,7 +4295,7 @@ inline void gcode_M109() {
     #define TEMP_CONDITIONS (!residency_start_ms || PENDING(now, residency_start_ms + (TEMP_RESIDENCY_TIME) * 1000UL))
   #else
     // Loop until the temperature is very close target
-    #define TEMP_CONDITIONS (isHeatingHotend(target_extruder))
+    #define TEMP_CONDITIONS (no_wait_for_cooling ? isHeatingHotend(target_extruder) : isCoolingHotend(target_extruder))
```

Comment:
This condition is unfortunately flawed:
```cpp
#define TEMP_CONDITIONS (no_wait_for_cooling ? isHeatingHotend(target_extruder) : isCoolingHotend(target_extruder))
```
…because `no_wait_for_cooling` means `S<any>`. Earlier in the code, we bailed if `S<lower>`, so at this point `no_wait_for_cooling` does guarantee `S<higher>`. However, `!no_wait_for_cooling` leaves open `R<higher>` as well as `R<lower>` so the last part of the ternary `isCoolingHotend(target_extruder)` is not the right condition to keep waiting.

I suggest instead the following simple condition, since it will work for `S<lower>` and `R<any>`:
```cpp
#define TEMP_CONDITIONS (fabs(degTargetHotend(target_extruder) - degHotend(target_extruder)) < 0.75f)
```